### DISCUSSION TITLE
Support of f64 for parser.try_parse

### DISCUSSION THREAD

**Discussion by heestree** [n4r1b/ferrisetw#94] [Q&A] [answered]
Hi 

Is there any plan to support f64 type for parser.try_parse? For instance, I need to get left, top, right, bottom (f64 type) of a rectangle in ETWGUID_DIRTYREGIONEVENT events from Microsoft-Windows-Dwm-Core provider. With this support, we can get a coordination of UI element from etl file and it would be very useful for UI analysis.

`let left: Option<f64> = parser.try_parse("left").ok();`
above line gives an syntax error "required by this bound in `Parser::<'schema, 'record>::try_parse`"
I believe this is due to the lack of floating point support for parser.try_parse impl.

**Comment by daladim** [comment]
Hello,

you are right, nothing is implemented in `parser.rs` to support `f32` or `f64` values.
If you'd like, go ahead and make a Pull Request, I'd be happy to review it! It should be fairly easy to do. Maybe it is as easy (to be confirmed)  as adding a `impl_try_parse_primitive!(f32)` and `impl_try_parse_primitive!(f64)`.

**Comment by heestree** [comment, marked as answer]
Hi @daladim yes, you're right.  I tested f32 and it worked!. Thanks.

In parser.rs
```
impl_try_parse_primitive!(f32);
impl_try_parse_primitive!(f64);
```
My codes:
```
if schema.provider_name() == "Microsoft-Windows-Dwm-Core" {
    if schema.task_name() == "ETWGUID_DIRTY_ADDRECT" {
        if schema.opcode_name() == "Info " {
            let parser = Parser::create(record, &schema);
            let left: Option<f32> = parser.try_parse("left").ok();
            if left.is_some() {
                dbg!(left);
            }
        } else {
            dbg!(schema.opcode_name());
        }
    }
}
```
output
```
left = Some(
    1030.0,
)
```
